src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -154,6 +158,29 @@
  *   String firstXML; // violation, XML abbreviation is not allowed
  * }
  * </pre>
+ * <p>
+ * To configure to check variables, enforcing no abbreviations
+ * except for variables that are both static and final (e.g. constants).
+ * </p>
+ * <p>Configuration:</p>
+ * <pre>
+ * &lt;module name="AbbreviationAsWordInName"&gt;
+ *     &lt;property name="tokens" value="VARIABLE_DEF"/&gt;
+ *     &lt;property name="ignoreFinal" value="false"/&gt;
+ *     &lt;property name="ignoreStatic" value="false"/&gt;
+ *     &lt;property name="ignoreStaticFinal" value="true"/&gt;
+ *     &lt;property name="allowedAbbreviationLength" value="0"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyClass {
+ *     public int AAAA = 1;              // violation, not ignored
+ *     public final int BBBB = 2;        // violation, not ignored
+ *     public static int CCCC = 3;       // violation, not ignored
```

Comment:
`violation, not ignored` -> just `violation` for all 3 lines above

Reply:
Fixed